packages/driver/cypress/integration/commands/net_stubbing_spec.ts

```diff
@@ -2685,6 +2685,30 @@ describe('network stubbing', { retries: 2 }, function () {
       .wait('@get')
     })
 
+    // https://github.com/cypress-io/cypress/issues/17084
+    it('does not overwrite the json-related content-type header', () => {
+      cy.intercept('/json-content-type', (req) => {
+        req.on('response', (res) => {
+          res.send({
+            statusCode: 500,
+            headers: {
+              'content-type': 'application/problem+json',
+              'access-control-allow-origin': '*',
+            },
+            body: {
+              status: 500,
+              title: 'Internal Server Error',
+            },
+          })
+        })
+      })
+
+      return fetch('/json-content-type')
+      .then((res) => {
+        expect(res.headers.get('content-type')).to.eq('application/problem+json')
+      })
```

Comment:
```suggestion
      cy.intercept('/json-content-type', (req) => {
        req.on('response', (res) => {
          res.send({
            statusCode: 500,
            headers: {
              'content-type': 'application/problem+json',
              'access-control-allow-origin': '*',
            },
            body: {
              status: 500,
              title: 'Internal Server Error',
            },
          })
        })
      })
      .then(() => {
        return fetch('/json-content-type')
        .then((res) => {
          expect(res.headers.get('content-type')).to.eq('application/problem+json')
        })
      })
```

The test should be written like above. Otherwise, Firefox and others shows a warning.